static/js/compose_validate.js

```diff
@@ -64,6 +65,41 @@ export function needs_subscribe_warning(user_id, stream_id) {
     return true;
 }
 
+export function warn_if_pm_mentions_user(mention_text, full_name, mentioned_user_id) {
+    if (compose_state.get_message_type() !== "private") {
+        return mention_text;
+    }
+    const recipient = compose_state.private_message_recipient();
+    const emails = util.extract_pm_recipients(recipient);
+    const to_user_ids = people.email_list_to_user_ids_string(emails);
+    if (to_user_ids === mentioned_user_id.toString()) {
+        const warning_area = $("#compose_pm_mentions_user_alert");
+        const context = {full_name};
+        const new_row = render_compose_pm_mentions_user_alert(context);
+
+        const user_mention_rx = new RegExp("@[*][*]" + full_name + "([|].*?)?[*][*]", "g");
+        const replace_text = "@**_" + full_name + "|" + mentioned_user_id + "**";
+        mention_text = mention_text.replace(user_mention_rx, replace_text);
+        // also replace other mentions...
+        compose_ui.replace_syntax(user_mention_rx, replace_text);
+        warning_area.data("full-name", full_name);
+        warning_area.data("mentioned-user-id", mentioned_user_id);
+        warning_area.append(new_row);
+        warning_area.show();
+    }
+    return mention_text;
+}
+
+export function undo_warn_if_pm_mentions_user(full_name, mentioned_user_id) {
+    if (compose_state.get_message_type() !== "private") {
+        return;
+    }
+    const user_mention_rx = new RegExp("@[*][*]_" + full_name + "([|].*?)?[*][*]", "g");
+    const replace_text = "@**" + full_name + "|" + mentioned_user_id + "**";
```

Comment:
I think we can directly use `people.get_mention_syntax` here. Also we use `user_id` in syntax only if there are duplicate names, so using the function will handle that too.

Reply:
oh sweet! much nicer - I'm on it (replacing now).